frontend/src/components/grantaccess/GrantAccess.vue

```diff
@@ -457,7 +463,7 @@ function removeForestClientFromList(index: number) {
                             style="width: 100% !important"
                             v-bind="field.value"
                             @update:modelValue="handleChange"
-                            @change="resetForestClientNumberData()"
+                            @change="removeForestClientSection()"
```

Comment:
we need to remove the whole forest client section when the role changed, include the forest client number input field and the forest client card